### PR TITLE
SECURITY: extended docs

### PR DESCRIPTION
**Pull request by bagder**
Also clarify BUG-BOUNTY.md with IBB details.

### CODE REVIEW

File: docs/BUG-BOUNTY.md [hunk 1]
```diff
@@ -11,26 +11,19 @@ HackerOne program](https://hackerone.com/curl).
...
-this program.
-
-See all details at [https://hackerone.com/curl](https://hackerone.com/curl)
-
-This bounty is relying on funds from sponsors. If you use curl professionally,
-consider helping fund this! See
-[https://opencollective.com/curl](https://opencollective.com/curl) for
-details.
+this program. See the [SECURITY-PROCESS](SECURITY-PROCESS.md) document for how
+we work on security issues.
```
Comment (danielgustafsson) [inline docs/BUG-BOUNTY.md:15]: s/on/with/ ?

File: docs/SECURITY-PROCESS.md [hunk 2]
```diff
@@ -119,23 +115,75 @@ Publishing Security Advisories
...
+## Bug Bounty
 
 See [BUG-BOUNTY](https://curl.se/docs/bugbounty.html) for details on the
 bug bounty program.
+
+# Not security issues
+
+This is an incomplete list of issues that are not considered vulnerabilities.
+
+## Small memory leaks
```
Comment (danielgustafsson) [inline docs/SECURITY-PROCESS.md:141]: What if the memory leaked contains a private key (or parts of one)? Should we clarify that we mean small leaks of insensitive data?

File: docs/BUG-BOUNTY.md (hunk 1 above)
Comment (bagder) [reply docs/BUG-BOUNTY.md]: :+1:

File: docs/SECURITY-PROCESS.md (hunk 2 above)
Comment (bagder) [reply docs/SECURITY-PROCESS.md:141]: amended accordingly

**Pull request closed without merging**